index.js

```diff
@@ -82,6 +82,25 @@ function addTests(transform, testDirectory, test) {
         assertEqual(template(locals).trim(), expected);
       }
     }
+    function checkFunctionAsyncOutput(template) {
+      if ((dependencies && dependencies.length) || (typeof template === 'object' && template)) {
+        assert(typeof template === 'object' && template, ' template should be an object because this module tracks dependencies');
+        assert(typeof template.fn === 'function', 'template.fn should be a function');
+        template.fn(locals).then(function (result) {
+          assertEqual(result.trim(), expected);
+        })
```

Comment:
This won't work. You have to return the promise to the caller in order to pass down any errors to the testing framework. Otherwise, the error just gets thrown "globally" (i.e. `setImmediate(function () { throw new Error() })`) and testit won't recognize the error.